### PR TITLE
fix: Upgrade eyes-cypress to latest

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
Fixes #27184

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N.A.

### TESTING INSTRUCTIONS
CI should pass

### ADDITIONAL INFORMATION

- [x] Has associated issue: #27184
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Hmmm... we need to check the license on this, and I'm having a weirdly hard time doing so.

**Comment by mistercrunch** [comment]
Screenshot 2024-04-16 at 4 09 44 PM
Screenshot 2024-04-16 at 4 09 59 PM

' this a recursive LICENSE!?

**Comment by mistercrunch** [comment]
puzzlin'
Screenshot 2024-04-16 at 4 12 25 PM
Screenshot 2024-04-16 at 4 12 21 PM

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -24,4 +24,6 @@ jobs:
           # find SPDX identifiers here: https://spdx.org/licenses/
           deny-licenses: MS-LPL, BUSL-1.1, QPL-1.0, Sleepycat, SSPL-1.0, CPOL-1.02, AGPL-3.0, GPL-1.0+, BSD-4-Clause-UC, NPL-1.0, NPL-1.1, JSON
           # adding an exception for an ambigious license on store2, which has been resolved in the latest version. It's MIT: https://github.com/nbubna/store/blob/master/LICENSE-MIT
-          allow-dependencies-licenses: 'pkg:npm/store2@2.14.2'
+          # adding exception for eyes-cypress, which has an explicit OSS license approved by ASF
+          #    license: https://applitools.com/legal/open-source-terms-of-use/
+          allow-dependencies-licenses: 'pkg:npm/store2@2.14.2, pkg:npm/applitools/eyes-cypress'
```
Comment (rusackas) [inline .github/workflows/dependency-review.yml:29]:
```suggestion
          allow-dependencies-licenses: 'pkg:npm/store2@2.14.2, pkg:npm/applitools/*'
```

**Review by rusackas — APPROVED** [review]

**Comment by geido** [comment]
@mistercrunch yeah that's quite terrible. However, Applitools confirmed that for our case they are applying [this license](https://applitools.com/legal/open-source-terms-of-use/#:~:text=Applitools%20hereby%20grants%20you%20a,open%20source%20project(s)). From a conversation with Apache legal, it seems we should be ok with this.

**Pull request merged**